src/core/sock/sockinfo.cpp

```diff
@@ -1968,53 +1962,48 @@ int sockinfo::modify_ratelimit(dst_entry *p_dst_entry, struct xlio_rate_limit_t
 
 int sockinfo::get_rings_num()
 {
-    int count = 0;
+    size_t count = 0;
     size_t num_rx_channel_fds;
-    if (is_socketxtreme()) {
-        /* socketXtreme mode support just single ring */
-        return 1;
+
+    ring *tx_ring = m_p_connected_dst_entry ? m_p_connected_dst_entry->get_ring() : nullptr;
+    if (tx_ring) {
+        (void)tx_ring->get_rx_channel_fds(count);
     }
 
-    rx_ring_map_t::iterator it = m_rx_ring_map.begin();
-    for (; it != m_rx_ring_map.end(); ++it) {
-        (void)it->first->get_rx_channel_fds(num_rx_channel_fds);
-        count += (int)num_rx_channel_fds;
+    for (auto pair : m_rx_ring_map) {
+        if (tx_ring == pair.first) {
+            continue;
+        }
+        (void)pair.first->get_rx_channel_fds(num_rx_channel_fds);
+        count += num_rx_channel_fds;
     }
-    return count;
+    return static_cast<int>(count);
 }
 
-int *sockinfo::get_rings_fds(int &res_length)
+int sockinfo::get_rings_fds(int *ring_fds, int ring_fds_sz)
 {
-    res_length = 0;
-    int index = 0;
     size_t num_rx_channel_fds;
+    int *channel_fds;
+    int index = 0;
 
-    if (is_socketxtreme()) {
-        /* socketXtreme mode support just single ring */
-        res_length = 1;
-        return m_p_rx_ring->get_rx_channel_fds(num_rx_channel_fds);
-    }
-
-    res_length = get_rings_num();
-    if (m_p_rings_fds) {
-        return m_p_rings_fds;
+    ring *tx_ring = m_p_connected_dst_entry ? m_p_connected_dst_entry->get_ring() : nullptr;
+    if (tx_ring) {
+        channel_fds = tx_ring->get_rx_channel_fds(num_rx_channel_fds);
```

Comment:
This will return m_p_rx_comp_event_channel. Do not we want the m_p_tx_comp_event_channel and use that as fd to poll ring TX?

Reply:
This fd is used in `xlio_socketxtreme_poll()` to identify the ring. It doesn't matter whether this is RX or TX cq. However, taking always the same cq type simplifies the checks, because we can simply make an array of unique fds. With TX cq we would need to handle different fds of the same ring.

Besides SocketXtreme, `xlio_socket_get_rings_fsd()` is mainly used in tests to compare values across different sockets.

Reply:
Lets then add a comment here that in future others don't think it is a bug.

Reply:
Done.